sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -97,7 +101,7 @@ case class Not(child: Expression)
 /**
  * Evaluates to `true` if `list` contains `value`.
  */
-case class In(value: Expression, list: Seq[Expression]) extends Predicate with CodegenFallback {
+case class In(value: Expression, list: Seq[Expression]) extends Predicate {
```

Comment:
It's better to use ExpectedInputType, see https://github.com/apache/spark/pull/7949/files#diff-9853dcf5ce3d2ac1e94d473197ff5768R136